### DISCUSSION TITLE
Is it possible to create a derived RwSignal?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [leptos-rs/leptos#2354] [Q&A] [answered]
Is there something like `Signal::derive` but for a `WriteSignal` or `RwSignal`?

**Comment by gbj** [comment, marked as answer]
You can get a mapped signal setter with [`SignalSetter`](https://docs.rs/leptos/latest/leptos/struct.SignalSetter.html), which is the "write only" equivalent to the "read only" `Signal` wrapper. There's no mapped RwSignal (i.e., no way to get from an `RwSignal<SomeStruct>` to `RwSignal<SomeStructField>`, although you can use `create_slice` to get a `Signal` and a `SignalSetter` for a single field of a struct)
